App/src/main/java/com/google/sps/servlets/ShelterMatch.java

```diff
@@ -49,41 +53,116 @@ public void addShelters(){
         nameFive.setPhoneNum(2137448186L);
         nameFive.setDescription("9-18 month program for veterans with mental health issues");
         nameFive.setAgeRange(18,100);
+        nameFour.setDisability(false);
+        list.add(nameFive);
+
+        Shelter nameSix = new Shelter();
+        nameSix.setName("Downtown Women’s Center: Permanent Supportive Housing");
+        nameSix.setLocation("442 S. San Pedro Street, Los Angeles, CA 90013");
+        nameSix.setPhoneNum(2136800600L);
+        nameSix.setDescription("Permanent program for homeless women.");
+        nameSix.setAgeRange(18,100);
+        nameSix.setGender("female");
+        nameFour.setDisability(false);
+        list.add(nameSix);
+
+        Shelter nameSeven = new Shelter();
+        nameSeven.setName("Downtown Women’s Center: Community-Based Housing Program");
+        nameSeven.setLocation("442 S. San Pedro Street, Los Angeles, CA 90013");
+        nameSeven.setPhoneNum(2136800600L);
+        nameSeven.setDescription("Nightly program for homeless women and children.");
+        nameSeven.setAgeRange(18,100);
+        nameSeven.setGender("female");
+        nameFour.setDisability(false);
+        list.add(nameSeven);
+
+        Shelter nameEight = new Shelter();
+        nameEight.setName("Higher Goals");
+        nameEight.setLocation("10510 S Vermont Ave, Los Angeles, CA - 90044");
+        nameEight.setPhoneNum(3237559702L);
+        nameEight.setDescription("6 month program to get you back on your feet.");
+        nameEight.setAgeRange(18,100);
+        nameEight.setDisability(false);
         list.add(nameFive);
     }
 
-    public static void sortShelters(Shelter[] shelters){
-        for(int i = 0; i < shelters.length; i++){
+    //returns a Shelter[] of the sorted results based on score
+    public static Shelter[] sortTop3(){
+        Shelter[] output = new Shelter[3];
+        ArrayList<Shelter> temp = new ArrayList<Shelter>();
 
+        //populating temp with Shelters from list so temp can be altered without effecting list
+        for(int i = 0; i < list.size(); i++){
+            temp.add(list.get(i));
         }
+        
+        int count = 0;
+        while(count <= output.length-1){
+            int posMax = 0;
+            int maxScore = temp.get(0).getScore();
 
+            for(int j = 0; j < temp.size(); j++){
+                if(maxScore < temp.get(j).getScore()){
+                    maxScore = temp.get(j).getScore();
+                    posMax = j;
+                }
+                else if(maxScore == temp.get(j).getScore()){
+                    maxScore = maxScore;
```

Comment:
What does this elseif section do? Why is it needed?

Reply:
The else if is mainly to make sure there is a Shelter eventually being added to the ArrayList. When I was thinking of scenarios to code for, I ran into the issue of if two different shelters contained the same score. Since the requirements for the shelters aren't all that different this would become a common issue. I decided at the time to make the original shelter holding the title of "maxScore" keep that title instead of the shelter it was being compared to becoming the new "maxScore". I also eventually forgot about it but if you have any suggestions on how to fix this I would love to hear them. My current code is the definitely me trying to force a shortcut with things. :)